### DISCUSSION TITLE
Cannot identify block in the document while dragging him

### DISCUSSION THREAD

**Discussion by gustavohleal** [GrapesJS/grapesjs#3747] [Q&A]
Hi Artf!

We've testing  some e2e libraries with grapes and getting some problems with the drag and drop funcionality.
The main problem is to identify a block that is being dragged into the canvas so we can manipulate the events with our tools.

I would like to ask you how exactly the drag and drop from grapes works, what events are fired and what elements are created when dragging some block from the moment someone select a block in the block manager and begin to drag.

Could you please explain that to us? Only found the code for dragging elements that already exists inside the canvas.
And if you could point us to the code that controls this events would be great.

Thank you

**Comment by artf** [comment]
it's native HTML5 drag and drop, here you can find how it binds to the Canvas https://github.com/artf/grapesjs/blob/dev/src/utils/Droppable.js
